include/deal.II/base/tensor.h

```diff
@@ -1945,25 +1940,6 @@ inline DEAL_II_CONSTEXPR DEAL_II_ALWAYS_INLINE
   return tmp;
 }
 
-/**
- * Entrywise multiplication of two tensor objects of rank 1.
- *
- * @relatesalso Tensor
- */
-template <int dim, typename Number, typename OtherNumber>
-inline DEAL_II_CONSTEXPR DEAL_II_ALWAYS_INLINE
-                         Tensor<1, dim, typename ProductType<Number, OtherNumber>::type>
-                         schur_product(const Tensor<1, dim, Number> &     src1,
-                                       const Tensor<1, dim, OtherNumber> &src2)
-{
-  Tensor<1, dim, typename ProductType<Number, OtherNumber>::type> tmp(src1);
-
-  for (unsigned int i = 0; i < dim; ++i)
-    tmp[i] *= src2[i];
-
-  return tmp;
```

Comment:
You can't remove this function -- it builds the base case of the recursion used in the following function.

Reply:
This is really unfortunate, as 
 - base case rank = 0
 - base case rank = 1
 - inductive step

is really not appealing (we have a one step recursion, and not a two step recursion). This redundancy was one of the reasons why I cleaned up the Tensor class a while ago. I will think about how to fix the underlying type issue for a minute.

Reply:
I have changed the second function variant (by explicitly forcing a `Tensor<rank-1, dim, Number>` intermediate. With this minor change `tensors/tensor_10` passes.

Reply:
Oh, I hadn't seen that there was another base case with rank=0...

Reply:
Well, you rightfully pointed out that this change (without the fix I commited later) breaks the test :smile: